### PR TITLE
CC-1317: serviced wait for network to come up

### PR DESCRIPTION
**Pull request by kwalkerz**
In our systemctl unit definition we wait for network-online.target, but
that condition appears to be satisfied when any non-loopback interface
gets an IP address?  What we constantly see in our log is docker0 has an
IP address but the NIC card does not, and our code panics.

Instead of panicing here, we'll loop (every second) waiting to be able
to obtain an outgoing IP address.  Terminate after an arbitrary amount
of time (90 seconds).

### CODE REVIEW

**Comment by kwalkerz** [comment]
retest this

**Comment by kwalkerz** [comment]
retest this

File: cli/api/utils.go [hunk 1]
```diff
@@ -41,11 +44,19 @@ func GetAgentIP(defaultRPCPort int) string {
...
 		return options.Endpoint
 	}
-	agentIP, err := utils.GetIPAddress()
-	if err != nil {
-		panic(err)
+
+	var agentIP string
+	var err error
+	for i := 1; i <= networkAccessAttempts; i++ {
+		if agentIP, err = utils.GetIPAddress(); err == nil {
```
Comment (wwwtyro) [inline cli/api/utils.go]: Does this work for offline installs? (Did it ever?)

Comment (jplouis) [inline cli/api/utils.go]: Look at line 44, I believe you need to set options when doing offline installs.

Comment (kwalkerz) [inline cli/api/utils.go]: If SERVICED_OUTBOUND_IP is set, then options.Endpoint will not be empty and offline install will work, right? Or is SERVICED_OUTBOUND_IP for something else?

Comment (jplouis) [inline cli/api/utils.go]: Ken, I think you are correct. My comment was to Rye.

Comment (wwwtyro) [inline cli/api/utils.go]: That's what Summer told me too, so sounds good.

Comment (kwalkerz) [inline cli/api/utils.go]: @JP--I suspected, but feeling very much out of the loop so trying to cover all bases. :-)

**Pull request merged**